### PR TITLE
SUS-4772 | Remove data for renamed user

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/SUS-4772
@Wikia/sus

### CODE REVIEW

File: extensions/wikia/Privacy/RemoveUserDataController.php [hunk 1]
```diff
@@ -77,6 +86,17 @@ private function getUserWikis( int $userId ) {
 		return $specialsDbr->selectFieldValues( 'events_local_users', 'wiki_id', ['user_id' => $userId], __METHOD__, ['DISTINCT'] );
 	}
 
+	private function getFakeUser( $username ) {
```
Comment (macbre) [inline extensions/wikia/Privacy/RemoveUserDataController.php]: What does `fake` is this context mean?

Comment (latata) [reply extensions/wikia/Privacy/RemoveUserDataController.php]: This is how the user, which is created during rename process, is named in the code:
https://github.com/Wikia/app/blob/25b19bc52b237f2c74d8d5a07d4d3e57c2228f37/extensions/wikia/UserRenameTool/RenameUserProcess.class.php#L443-L443

Comment (latata) [reply extensions/wikia/Privacy/RemoveUserDataController.php]: I can add comment about it

File: extensions/wikia/Privacy/RemoveUserDataController.php [hunk 2]
```diff
@@ -28,18 +29,28 @@ public function removeUserData() {
 
 		$this->info( "Right to be forgotten request for user $userId", ['userId' => $userId] );
 
+		$username = User::whoIs( $userId );
+		$fakeUser = $this->getFakeUser($username);
```
Comment (pwojcik86) [inline extensions/wikia/Privacy/RemoveUserDataController.php:33]: Do we need the whole user object here? Won't `user_id` be enough?

File: extensions/wikia/Privacy/RemoveUserDataOnWikiTask.php [hunk 3]
```diff
@@ -120,14 +120,18 @@ private function removeUserPagesFromRecentChanges( $userDbKey ) {
...
 		$this->removeCheckUserData( $userId );
 		$this->removeAbuseFilterData( $userId );
 		$this->removeIpFromRecentChanges( $userId );
 
 		$userDbKey = Title::newFromText( $username )->getDBkey();
 		$this->removeUserPages( $userDbKey );
 		$this->removeUserPagesFromRecentChanges( $userDbKey );
+		if ( !empty( $oldUsername ) ) {
+			$oldUserDbKey = Title::newFromText( $oldUsername )->getDBkey();
+			$this->removeUserPages( $oldUserDbKey );
```
Comment (pwojcik86) [inline extensions/wikia/Privacy/RemoveUserDataOnWikiTask.php]: We also need `$this->removeUserPagesFromRecentChanges( ... );`

**Review by pwojcik86 — APPROVED** [review]

**Pull request merged**